client/task_runner.go

```diff
@@ -288,6 +284,31 @@ func NewTaskRunner(logger *log.Logger, config *config.Config,
 		},
 	}
 
+	if tc.alloc.Job.ParentID != "" {
+		tc.baseLabels = append(tc.baseLabels, metrics.Label{
```

Comment:
Discussed this with the team internally, and we don't want to break any existing users that rely on the full jobID for dispatched jobs. Could you add a new "parentJob"  label here and below for alloc.job.ParentID and leave the existing "job" label as it is? That way it doesn't break backwards compatibility.